pandas/core/frame.py

```diff
@@ -2892,14 +2892,17 @@ def _get_value(self, index, col, takeable: bool = False):
         engine = self.index._engine
 
         try:
-            return engine.get_value(series._values, index)
+            if isinstance(series._values, np.ndarray):
```

Comment:
i understand why you are doing this (does it affect perf btw?)

but is not a great long term pattern. can you instead factor this to a function (which you can also use in Series)?

Reply:
long-term this is going to be made unnecessary when Series._values is changed to match Index._values.  At that point libindex.get_value_at and IndexEngine.get_value will become entirely unnecessary

Reply:
k sounds like a plan!